### PR TITLE
delete customer unit test case

### PR DESCRIPTION
**Pull request by sathish199308**
Delete Customer Unit Test Case

### CODE REVIEW

File: BookingSystem.Tests/Integration.Tests/CustomerRepositoryTest/DeleteCustomerTest.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        {
+            //Arrange
+            var customerData = fixture.Build<CustomerVM>().With(a => a.IsDeleted, false)
+              .With(x => x.CustomerID, 1).Create();
+            var customer = _testmapper.Map<Customer>(customerData);
+            var exceptedCustomerId = customer.CustomerID;
+            var options = new DbContextOptionsBuilder<DataContext>().UseInMemoryDatabase(databaseName: "BookingSystem").Options;
+
+            using (var context = new DataContext(options))
+            {
```
Comment (Backiaraj07) [inline BookingSystem.Tests/Integration.Tests/CustomerRepositoryTest/DeleteCustomerTest.cs:59]: Dont hardcode the values, refer previous methods

File: BookingSystem/Controllers/BookingController.cs [hunk 2]
```diff
@@ -71,6 +71,15 @@ public async Task<IActionResult> Index()
...
 
+       
+
+        #region Getting the Booking Record for Save
+
+        /// <summary>
+        /// Getting the Booking Record for Save
+        /// </summary>
+        /// <returns>Getting the Booking Record for Save</returns>
+
```
Comment (Backiaraj07) [inline BookingSystem/Controllers/BookingController.cs:82]: As per our requirement, we dont need to add Summary

File: BookingSystem/Controllers/CustomerController.cs [hunk 3]
```diff
@@ -82,5 +82,15 @@ public async Task<IActionResult> Customer(int? CustomerID)
...
+        [HttpGet]
+        public async  Task<IActionResult> DeleteCustomer(int? CustomerID)
+        {
+            if (CustomerID > 0)
+            {
+                await _clientCustomerRepository.DeleteCustomer(CustomerID ?? 0);
+            }
+            return RedirectToAction("Index");
+        }
     }
```
Comment (Backiaraj07) [inline BookingSystem/Controllers/CustomerController.cs:95]: Use nameof fuction to redirecttoAction

**Review by Backiaraj07 — COMMENTED** [review]
Comments added, Kindly fix it.

File: BookingSystem.DAL/Repository/CustomerRepository.cs [hunk 4]
```diff
@@ -105,7 +105,15 @@ public async Task<CustomerVM> GetActiveCustomerByCustomerIdAsync(int? CustomerID
        
         public async Task<bool> DeleteCustomerAsync(int CustomerID)
         {
-            throw new System.NotImplementedException();
+            var bookingInfo = _dataContext.Customer.FirstOrDefault(a => a.CustomerID == CustomerID);
+            bookingInfo.IsDeleted = true;
+            var results = _dataContext.SaveChanges();
```
Comment (manic1990) [inline BookingSystem.DAL/Repository/CustomerRepository.cs:110]: Use SaveChangesAsync();

File: BookingSystem/Controllers/CustomerController.cs [hunk 5]
```diff
@@ -82,5 +82,15 @@ public async Task<IActionResult> Customer(int? CustomerID)
 
             return View();
         }
+
+        [HttpGet]
+        public async  Task<IActionResult> DeleteCustomer(int? CustomerID)
+        {
+            if (CustomerID > 0)
+            {
+                await _clientCustomerRepository.DeleteCustomer(CustomerID ?? 0);
```
Comment (manic1990) [inline BookingSystem/Controllers/CustomerController.cs:91]: Use ConfigureAwait(false);

File: BookingSystemAPI/Controllers/CustomerController.cs [hunk 6]
```diff
@@ -117,10 +118,40 @@ public async Task<ActionResult> GetCustomerById(int? customerId)
...
+        public async Task<IActionResult> DeleteCustomer(int CustomerID)
+        {
+            _logger.LogInformation(nameof(CustomerController) + nameof(DeleteCustomer), started);
+            var response = new ResponseModel();
+            if (CustomerID > 0)
+            {
+                try
+                {
+                    _logger.LogInformation(nameof(CustomerService) + deletecustomerbyId, started);
+                    var result = _customerService.DeleteCustomer(CustomerID);
```
Comment (manic1990) [inline BookingSystemAPI/Controllers/CustomerController.cs:135]: Use ConfigureAwait(false)

File: BookingSystem.Tests/Integration.Tests/CustomerRepositoryTest/DeleteCustomerTest.cs (hunk 1 above)
Comment (sathish199308) [reply BookingSystem.Tests/Integration.Tests/CustomerRepositoryTest/DeleteCustomerTest.cs:59]: Hard Coded Value is removed

File: BookingSystem/Controllers/BookingController.cs (hunk 2 above)
Comment (sathish199308) [reply BookingSystem/Controllers/BookingController.cs:82]: Comments Removed

File: BookingSystem/Controllers/CustomerController.cs (hunk 3 above)
Comment (sathish199308) [reply BookingSystem/Controllers/CustomerController.cs:95]: Nameoffunction is implemented

File: BookingSystem.DAL/Repository/CustomerRepository.cs (hunk 4 above)
Comment (sathish199308) [reply BookingSystem.DAL/Repository/CustomerRepository.cs:110]: Added

File: BookingSystem/Controllers/CustomerController.cs (hunk 5 above)
Comment (sathish199308) [reply BookingSystem/Controllers/CustomerController.cs:91]: Added

**Review by manic1990 — APPROVED** [review]

File: BookingSystemAPI/Controllers/HotelController.cs [hunk 7]
```diff
@@ -91,28 +89,57 @@ public async Task<ActionResult> Get(int HotelID)
...
         {
-            _logger.LogInformation(nameof(HotelController) + nameof(Post), started);
             var response = new ResponseModel();
             try
             {
-                _logger.LogInformation(nameof(HotelService) + saveHotel, started);
                 var result = await _hotelService.SaveHotelDetailsAsync(hotel);
 
                 response.statuscode = HttpStatusCode.Created;
                 response.errordescription = "";
```
Comment (Backiaraj07) [inline BookingSystemAPI/Controllers/HotelController.cs:98]: add configureawait(false)

File: BookingSystemAPI/Controllers/HotelController.cs [hunk 8]
```diff
@@ -91,28 +89,57 @@ public async Task<ActionResult> Get(int HotelID)
...
+            _logger.LogInformation(nameof(HotelController) + nameof(DeleteHotel), started);
+            var response = new ResponseModel();
+            if (hotelId > 0)
+            {
+                try
+                {
+                    _logger.LogInformation(nameof(CustomerService) + deletehotelbyId, started);
+                    var result = _hotelService.DeleteHotelDetailsAsync(hotelId);
+
+                    response.statuscode = HttpStatusCode.Created;
```
Comment (Backiaraj07) [inline BookingSystemAPI/Controllers/HotelController.cs:124]: add configureawait(false)

**Review by Backiaraj07 — COMMENTED** [review]
comments added

**Review by Backiaraj07 — APPROVED** [review]
Comment fixed.

**Comment by Backiaraj07** [comment]
PR verfied

**Pull request merged**